import "SRXDataClass.proto";
import "SRXDataSchool.proto";
import "SRXDataImage.proto";

message SRXDataUser {
  // only 1 of below could be specified.
  optional string id = 1;
}

// Request and Response of API CreateClass.

message SRXPrtoCreateSchoolRequest {
  // Required.
  optional SRXDataUser requesting_user = 1;

  // Required.
  optional SRXDataSchoolInfo school = 2;
}

message SRXProtoCreateSchoolResponse {
  optional string school_id = 1;
}

message SRXProtoCreateClassRequest {
  // Required.
  optional SRXDataUser requesting_user = 1;

  // Required.
  optional SRXDataClassInfo class_info = 2;
}

message SRXProtoCreateClassResponse {
  optional string class_id = 1;
}


// Request and Response of API ReadClass.
message SRXDataRoleInClassTypeEnum {
enum SRXDataRoleInClassType {
 UNKNOWN = 0;
 OWNER = 1;
 STUDENT = 2;
}
}

message SRXProtoReadClassRequest {
  // Required.
  optional SRXDataUser requesting_user = 1;

  // Required.
  optional SRXDataRoleInClassTypeEnum.SRXDataRoleInClassType role_in_class = 2 [default = UNKNOWN];
}

message SRXProtoReadClassResponse {
  repeated SRXDataClassInfo class_collection = 1;
}

// Request and Response of API SearchClass.
message SRXProtoSearchClassRequest {
  // Required.
  optional float latitude = 1;

  // Required.
  optional float longtitude = 2;

  // Required
  optional float radius_latitude = 3;

  // Required
  optional float radius_longtitude = 4;
}

message SRXProtoSearchClassResponse {
  repeated SRXDataClassInfo class_collection = 1;
}

// Request and Response of API addImages
message SRXProtoImage {
  optional bytes data = 1;
}

message SRXProtoAddImagesRequest {
  // Required.
  optional SRXDataUser requesting_user = 1;

  optional SRXDataImageServerTypeEnum.ImageServerType server_type = 2;

  repeated SRXDataImage image = 3;
}

message SRXProtoAddImagesResponse {
  // Should be 1:1 aligned with image_file.
  repeated string image_key = 1;
}

// Request ands Response of API getImages
message SRXProtoGetImagesRequest {
  // Required.
  optional SRXDataImageServerTypeEnum.ImageServerType server_type = 1;

  // Required.
  repeated string image_key = 2;
}

message SRXProtoGetImagesResponse {
  repeated SRXDataImage image_data = 1;
}



message SRXProtoReadProfileRequest {
   // Required
   optional SRXDataUser requesting_user = 1;
}

message SRXProtoGetProfileResponse {

}

message SRXProtoMutableProfileRequest {

}

message SRXProtoMutableProfileResponse {

}


enum SRXProtoLogInMethodEnum {
  DEFAULT = 0;
  NATIVE = 1;
}


message SRXProtoNativeLogInMessage {
  optional string user_name = 1;
  optional string password = 2;
}

message SRXProtoLogInRequest {
  // Required.
  optional SRXProtoLogInMethodEnum log_in_method = 1;

  // Required.
  optional SRXProtoNativeLogInMessage native_log_in = 2;
}

message SRXProtoLogInResponse {
  // Required.
  optional SRXDataUser user = 1;
}
